### ISSUE TITLE
design rationale for using `#` in keys

### ISSUE THREAD

**Issue by ghostsquad** [jsonnet-libs/xtd#2]
what was the design rationale for using keys that start with `#`? I don't think that's very idiomatic (or pretty) jsonnet. But of course, I don't have any specific guides that I can point to, just a lot of other jsonnet.

**Comment by sh0rez** [comment]
Hi, that's for a experimental, jsonnet specific docs generator called https://github.com/sh0rez/docsonnet. 

It uses the idea of structured docs, where docs are actually part of the code. This is done because:

- it is colocated to the code and easier to update
- it can be overwritten (patched) using standard jsonnet operations. When your final exposed API is a sum of patches, these need to be in the docs as well. I don't want to write a custom jsonnet evaluator that also ready comments and merges those.

I agree it's ugly and complex, but it's all I got right now

**Comment by sbarzowski** [comment]
> I agree it's ugly and complex, but it's all I got right now

Hmmm... it's not that bad IMO. And it sidesteps the biggest obstacle of the comment extraction approach – that it's sometimes hard to follow statically where the documented values come from. And on some level it doesn't matter, because from user's perspective the only thing that matters is the what a library evaluates to (and not if it was constructed directly or with some weird object comprehension which messes up the static analysis).

**Comment by ghostsquad** [comment]
OMG I totally missed that the actual function is on the next line. 🤦 

```
'#escapeString': ..
escapeString(str)::

```
